### PR TITLE
gdal: use pytest-benchmark to disable benchmarks

### PR DESCRIPTION
**Pull request by mweinelt**
This is a the reliable way to make sure we don't run any benchmarks,
since apparently disabling the benchmark directory has become
insufficient.

https://hydra.nixos.org/build/253065498/nixlog/2

cc #295458

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by imincik** [comment]
Wow, merged without any chance of review ? 

I understand that you are under heavy load with staging, but it would be great to give us at least one day for providing feedback.

File: pkgs/development/libraries/gdal/default.nix [hunk 1]
```diff
@@ -234,18 +234,19 @@ stdenv.mkDerivation (finalAttrs: {
   '';
   nativeInstallCheckInputs = with python3.pkgs; [
     pytestCheckHook
+    pytest-benchmark
```
Comment (imincik) [inline pkgs/development/libraries/gdal/default.nix:237]: Why adding pytest-benchmark to build inputs at the moment when we want to disable it ?

Comment (imincik) [reply pkgs/development/libraries/gdal/default.nix:237]: OK, I see, building without `pytest-benchmark` is causing

```
Executing pytestCheckPhase
/build/source/build/autotest /build/source/build
Copying /nix/store/kvpxdc0027xzkfwyi145r8v139wj24mf-proj-9.4.0/share/proj/proj.db to /build/source/build/autotest/gcore/tmp/proj_db_tmpdir
ERROR: usage: __main__.py [options] [file_or_dir] [file_or_dir] [...]
__main__.py: error: unrecognized arguments: --benchmark-disable
  inifile: /build/source/build/autotest/pytest.ini
  rootdir: /build/source/build/autotest
```.

But I would still prefer something less counter-intuitive.

**Comment by mweinelt** [comment]
> @mweinelt , Wow, merged without any chance of review ?

I saw your upvote and interpreted it as you being okay with this change. Feel free to improve it.

File: pkgs/development/libraries/gdal/default.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/development/libraries/gdal/default.nix:237]: The tests complained about the missing fixture, and pytest-benchmark kinda offers the best targeting mechanism to disable all benchmark tests.

I also tried modifying the `disabledTestPaths`, but with limited time I could not get it to work.

**Comment by vcunat** [comment]
I guess it can get confused with
> Add a 👍 [reaction](https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to [pull requests you find important](https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

File: pkgs/development/libraries/gdal/default.nix (hunk 1 above)
Comment (imincik) [reply pkgs/development/libraries/gdal/default.nix:237]: OK, thanks. For now, I can't find a better way thow to disable them either.

**Comment by imincik** [comment]
> > @mweinelt , Wow, merged without any chance of review ?
> 
> I saw your upvote and interpreted it as you being okay with this change. Feel free to improve it.

OK, so it was misunderstanding then. I did thumbs-up for you to know that I see PR and will review it ASAP.

**Comment by mweinelt** [comment]
Best to use :eyes: in that case, I guess.

**Pull request merged**